### DISCUSSION TITLE
HDR global toggle and Automatic LUT Calibration for Windows software screen grabber

### DISCUSSION THREAD

**Discussion by Switch01** [awawa-dev/HyperHDR#515] [General]
Hello, 

i'm using HyperHDR on my HDR monitor and Windows 11 computer, without any usb grabbers, and it works great! But i'm having issues with the global HDR toggle and the automatic calibration tool:

- The global toggle doesn't seem to apply to the software screen grabber "HDR to SDR tone mapping" option, according to the logs:
"(DxGrabber.cpp:109) setHdrToneMappingMode to: Fullscreen" this is what changes with that settings, but this is what changes instead with the global toggle: "(MFGrabber.cpp:175) setHdrToneMappingMode to: Fullscreen"

- The automatic LUT calibration tool gives the error "[MEDIA_FOUNDATION:AUTO] Could not find any capture device" after showing the pattern and creating the empty LUT and it doesn't seem to continue. (also, if i try to start it in Fullscreen mode (F11) it says "please enter fullscreen", it only starts in not-fullscreen mode).

Also, sort of unrelated to this, i'm using another software called "Novideo sRGB" to clamp my monitor colors in SDR mode, and this software is capable of detecting if HDR is active or not by using a Windows API to disable itself, would it be possible to implement something like this?

**Comment by awawa-dev** [comment]
Hi

Will check the DirectX software potential calibration issues later.

>  (also, if i try to start it in Fullscreen mode (F11) it says "please enter fullscreen", it only starts in not-fullscreen mode).

Please use the Firefox, best to use only single tab with HyperHDR panel. Fullscreen detection is non-standardized between web-browsers and currently only Firefox will work for sure in Windows 10 (Windows 11 is not supported by me, I'm counting here on the users). In the full-screen when the mosaic is shown, no browser menu may be visible: mosaic should take all the space.

> Also, sort of unrelated to this, i'm using another software called "Novideo sRGB" to clamp my monitor colors in SDR mode, and this software is capable of detecting if HDR is active or not by using a Windows API to disable itself, would it be possible to implement something like this?

Theoretically it's possible. As you can see in the detailed calibrations logs which are attached to the LUT tables, values for raw HDR content are usually in ~[0-150] RGB range. So it's a matter to implement it.

**Reply by Switch01** [reply]
I don't know why i haven't tought about using Firefox earlier, now the calibration goes fullscreen perfectly! (It still doesn't complete the calibration though)

**Comment by awawa-dev** [comment]
I submitted https://github.com/awawa-dev/HyperHDR/pull/518
You can download the installers from here (it will expire soon): https://github.com/awawa-dev/HyperHDR/actions/runs/4227197902 or compile it yourself. But on my system Windows already does the HDR tone mapping when software capturing is used...unfortunately in very aggressive way and this doesnt look good. Unless it's disabled somehow, the calibration procedure wont work properly because it requires raw HDR content (dark & washed-out colors).

**Reply by Switch01** [reply]
First of all thank you for the quick update, the calibration process now works perfectly and creates a LUT file!

But you are right, i didn't notice it before but as you said Windows applies a tonemapping to (some) HDR content that ruins the calibration process. The problem is that this tonemapping completely blows out the highlights, making everything too bright.

SDR:
 ![chrome_VSAb4z5J0T (Custom)](https://user-images.githubusercontent.com/48028783/220339799-006119af-705e-4abd-b107-aa5245bdd06c.jpg) 

HDR: 
![chrome_cNfxEsGKGX (Custom)](https://user-images.githubusercontent.com/48028783/220339870-eb5a639e-9874-47d8-ae87-3201d0767885.jpg)
 

I would need to find a way to make a calibrated LUT for this kind of tonemapping, if it's even possible. 

But there is also another problem, Windows doesn't apply this tonemapping to everything, for example while playing Elden Ring the grabber captures raw HDR, with washed out colors and brightness, so i would need to have two different LUT files.

![chrome_uUZw1VPC7c (Custom)](https://user-images.githubusercontent.com/48028783/220340446-b5d94a99-e057-4b44-affc-2085b9c0feae.jpg)

I don't think that there are any easy solutions for these issues,

Thanks again for the help!

**Reply by awawa-dev** [reply]
@Switch01 I made a progress and it should be fixed soon: https://github.com/awawa-dev/HyperHDR/issues/689
